src/lib.rs

```diff
@@ -268,7 +274,7 @@ impl<T: ?Sized> TakeCell<T> {
     /// [`heal`]: TakeCell::heal
     #[allow(clippy::mut_from_ref)]
     pub unsafe fn steal(&self) -> &mut T {
-        self.taken.store(true, Ordering::SeqCst);
+        self.taken.store(true, Ordering::Relaxed);
```

Comment:
> Guess this can be non-atomic at all.

There is no way to do this in Rust afaik

> Include into the doc comment that this call must be externally synchronized with `take` calls in other threads.

Because otherwise other threads can see the old (`false`) value while the thread calling `steal` has already got the reference? Or what exactly is the problem here?

> I'd double-check with debug_assertion that old value is indeed `false`

Right, I intended to add `debug_assert` here, but forgot. Thanks for reminding me.

Reply:
> > Guess this can be non-atomic at all.

> There is no way to do this in Rust afaik

There's unstable function. https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU8.html#method.as_mut_ptr



> > Include into the doc comment that this call must be externally synchronized with take calls in other threads.

> Because otherwise other threads can see the old (false) value while the thread calling steal has already got the reference? Or what exactly is the problem here?

Exactly. Even if other means signal that `steal` was called before `take` on other thread. Without memory fence the `take` call on other thread may see old `false`.

Reply:
I don't think that's really necessary though (I don't even have a use-case for `steal` to be honest, it exists just for API completeness). So I'll keep the current implementation.